File: Programs/LC 6/LC_6_1.py
import os
cwd = os.path.dirname(__file__)

def convert(s: str, numRows: int) -> str:
    if numRows == 1: return s
    
    rows = [''] * numRows

    rowNum, increasing = 1, True

    for ch in s:
        if rowNum == 1: increasing = True
        elif rowNum == numRows: increasing = False

        rows[rowNum - 1] += ch
        rowNum = (rowNum + 1) if increasing else (rowNum - 1)
    
    return ''.join(rows)

if __name__ == '__main__': 
    nInputs = 5

    for i in range(1, nInputs + 1): 
        fname = "input{number}.txt"
        fpath = os.path.join(cwd, fname.format(number = i))
        
        print("======== TestCase {number} ========".format(number = i))
        try: 
            with open(fpath, "r") as fileHandle: 
                #  ===== START =====
                s = fileHandle.readline().strip()
                numRows = int(fileHandle.readline().strip())

                print(convert(s, numRows))

                #  ===== END =====
        except FileNotFoundError:
            print("Cannot Open the test case")
            break

File: Programs/LC 6/LC_6_1.cpp
#include<iostream>
#include<fstream>
#include<vector>
#include<sstream>

using namespace std;

static int NeedForSpeed = []() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(NULL);
    std::cout.tie(0);
    return 0;
}();

string convert(string s, int numRows){
    if(numRows == 1) return s; 

    vector<string> rows(numRows); 

    bool increasing = true; 
    int rowNum = 1; 

    int n = s.length(); 
    for(int i = 0; i < n; i++){
        if(rowNum == 1) increasing = true; 
        else if(rowNum == numRows) increasing = false; 

        rows[rowNum - 1] += s[i];
        rowNum = (increasing)?(rowNum + 1):(rowNum - 1); 
    }

    string res = ""; 
    for(int i = 0; i < numRows; ++i)
        res += rows[i]; 
    

    return res; 
}

int main(){
    // change the number of test cases
    int nInputs = 5;

    // test case files should be input1.txt, input2.txt, ..., inputN.txt format
    for(int i = 1; i <= nInputs; i++){
        string filename = "input" + to_string(i) + ".txt"; 

        ifstream fin(filename); 
        if(!fin.is_open()){
            cout<<"Cannot Open the test case"<<endl;
            return 0; 
        }

        cout<<"======== TestCase "<<i<<" ========"<<endl;


        // Write input and output here
        // Instead of cin use fin

        // ===== START =====
        string s; fin>>s; 
        int numRows; fin>>numRows; 

        cout<<convert(s, numRows)<<endl; 
        
        // ===== END =====
    }
    
    return 0; 
}